build/js/main.js

```diff
@@ -1077,92 +1095,96 @@ class Bluetooth extends Plugin {
-                    console.error(err);
-                    return;
+    renderStatus () {
+        if (this.devStatus !== undefined) {
+            for (var i =0; i<this.devStatus.length;i++) {
+                if (this.devStatus[i].connected =="true") {
+                    document.getElementById('BT_Disconnect').disabled = false;
                 }
-
-                // bail out if the plugin returns nothing
-                if (resp === undefined || resp.DeviceList.length < 0)
-                    return;
-
-                this.discoveredDevices = resp.DeviceList;
-                this.renderDiscoveredDevices();
-            });
-        }
-    }
-
-    checkDeviceScanning() {
-        api.getPluginData(this.callsign, (err, resp) => {
-            if (resp.scanning) {
-                this.stopScan();
             }
-        });
+        }
+        this.scanningStatus.innerHTML = this.scanning === true ? 'ON' : 'OFF';
+        if(!this.scanning) {
+	    document.getElementById('BT_ScanForDevices').disabled = false;
+            clearInterval(this.Timer);
+	}
     }
 
-    /* ----------------------------- RENDERING ------------------------------*/
-
-    renderStatus () {
-        if (this.connected !== '') {
-            var deviceInfo = this.pairedDevices.find(deviceInfo=>deviceInfo.Address==this.connected);
-            this.connectedStatus.innerHTML = deviceInfo.Name;
-        } else
-            this.connectedStatus.innerHTML = 'Not Connected';
-
-        this.scanningStatus.innerHTML = this.scanning === true ? 'ON' : 'OFF';
+    renderDiscoveredDevices () {
+        this.discoveredDeviceList.innerHTML = '';
+        for (var i=0; i<this.discoveredDevices.length; i++) {
+            var newChild = this.discoveredDeviceList.appendChild(document.createElement("option"));
+            if (this.discoveredDevices[i].name === "")
+                newChild.innerHTML = `${this.discoveredDevices[i].address}`;
+            else
+                newChild.innerHTML = `${this.discoveredDevices[i].name}`;
+        }
     }
 
     renderPairedDevices () {
-        this.devicesListEl.innerHTML = '';
-        for (var i=0; i<this.pairedDevices.length; i++) {
-            var newChild = this.devicesListEl.appendChild(document.createElement("option"));
-            if (this.pairedDevices[i].Name === '')
-                newChild.innerHTML = `${this.pairedDevices[i].Address}`;
-            else
-                newChild.innerHTML = `${this.pairedDevices[i].Name}`;
+        this.pairedDeviceList.innerHTML = '';
+        if(this.discoveredDevices.length !==0){
+            document.getElementById('BT_Pair').disabled = false;
+        }
+        var j=0;
+        for (var i=0; i<this.discoveredDevices.length; i++) {
+            if(this.discoveredDevices[i].paired) {
+                var newChild = this.pairedDeviceList.appendChild(document.createElement("option"));
+                this.pairedDevices[j] = this.discoveredDevices[i];
```

Comment:
Does the pairedDevice array need to line up with a specific value? 

Else its easier to drop the `j=0` and `j = j+1` part and just use `this.pairedDevices.push( this.discoveredDevices[i] )` instead. Just make sure this.pairedDevices is initialized as an array `this.pairedDevices = []`